server/src/main/java/io/crate/metadata/RelationName.java

```diff
@@ -65,9 +65,21 @@ public static RelationName of(QualifiedName name, String defaultSchema) {
             ensureIsCrateCatalog(parts.get(0));
         }
         if (parts.size() >= 2) {
-            return new RelationName(parts.get(parts.size() - 2), parts.get(parts.size() - 1));
+            return RelationName.of(parts.get(parts.size() - 2), parts.get(parts.size() - 1));
         }
-        return new RelationName(defaultSchema, parts.get(0));
+        return RelationName.of(defaultSchema, parts.get(0));
+    }
+
+    public static RelationName of(@Nullable String schema, String name) {
+        assert name != null : "table name must not be null";
+        if (schema != null && containsIllegalCharacters(schema)) {
+            throw new InvalidSchemaNameException(schema);
+        }
+        if (containsIllegalCharacters(name)) {
+            var invalidRelationName = new RelationName(schema, name);
+            throw new InvalidRelationName(invalidRelationName);
+        }
```

Comment:
Shouldn't this run all checks? We still have `ensureValidForRelationCreation` which still wouldn't be used for swap so you could still create invalid relations with it?

Reply:
I thought since `swap tables` work with existing table only so it seemed not needed to check all. Anyways, I am consolidating these checks, I think it looks cleaner.

Reply:
I tried to consolidate all checks as much as possible. Also, FYI, schema names starting with '_' will be allowed with https://github.com/crate/crate/issues/12110#issuecomment-1108628045